packages/manager/apps/web-office/src/pages/dashboard/users/ModalEditUsers.component.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useParams, useSearchParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import {
+  OdsFormField,
+  OdsIcon,
+  OdsInput,
+  OdsText,
+  OdsTooltip,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_INPUT_TYPE,
+  ODS_MODAL_COLOR,
+  ODS_TEXT_PRESET,
+} from '@ovhcloud/ods-components';
+import { useForm, Controller } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import { useMutation } from '@tanstack/react-query';
+import { ApiError } from '@ovh-ux/manager-core-api';
+import { useGenerateUrl, useOfficeUserDetail } from '@/hooks';
+import Modal from '@/components/Modals/Modal';
+import queryClient from '@/queryClient';
+import { EDIT_USERS_FORM_SCHEMA } from '@/utils/formSchemas.utils';
+import {
+  getOfficeLicenseQueryKey,
+  putOfficeLicenseDetails,
+} from '@/api/license';
+import { getOfficeUsersQueryKey, putOfficeUserDetail } from '@/api/users';
+import { UserParamsType } from '@/api/api.type';
+
+export default function ModalEditUsers() {
+  const { t } = useTranslation(['dashboard/users/edit', 'common']);
+  const navigate = useNavigate();
+
+  const { serviceName: selectedServiceName } = useParams();
+  const [searchParams] = useSearchParams();
+  const activationEmail = searchParams.get('activationEmail');
+  const licencePrepaidName = searchParams.get('licencePrepaidName');
+
+  const { addError, addSuccess } = useNotifications();
+
+  const goBackUrl = useGenerateUrl('..', 'path');
+  const onClose = () => navigate(goBackUrl);
+
+  const {
+    data: userDetail,
+    isLoading: isUserDetailLoading,
+  } = useOfficeUserDetail(activationEmail, licencePrepaidName);
+
+  const [isLoading, setIsLoading] = useState(
+    !userDetail || isUserDetailLoading,
+  );
+
+  const {
+    control,
+    setValue,
+    getValues,
+    formState: { isDirty, isValid, errors },
+  } = useForm({
+    defaultValues: {
+      firstname: userDetail?.firstName || '',
+      lastname: userDetail?.lastName || '',
+      login: userDetail?.activationEmail.split('@')[0] || '',
```

Comment:
can activationEmail be undefined or null ?

Reply:
No it can't.